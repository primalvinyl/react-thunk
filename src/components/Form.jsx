import React from 'react';
import { Formik } from 'formik';
import { useDispatch } from 'react-redux';
import { userRequest } from '../actions';

const Form = props => {
    const dispatch = useDispatch();

    return (
        <Formik
            initialValues={{ screenName: props.screenName }}
            onSubmit={({ screenName }, { setSubmitting }) => {
                dispatch(userRequest(screenName));
                setSubmitting(false);
            }}
        >
            {({
                values,
                errors,
                touched,
                handleChange,
                handleBlur,
                handleSubmit,
                isSubmitting
            }) => (
                    <form onSubmit={handleSubmit}>
                        <input
                            type="text"
                            name="screenName"
                            onChange={handleChange}
                            onBlur={handleBlur}
                            value={values.screenName}
                        />
                        {errors.screenName && touched.screenName && errors.screenName}
                        <button type="submit" disabled={isSubmitting}>
                            Submit
                        </button>
                    </form>
                )}
        </Formik>
    );
}

export default Form;
