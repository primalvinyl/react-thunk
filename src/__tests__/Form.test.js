import React from 'react';
import Renderer from 'react-test-renderer';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import rootReducer from '../store/reducers';
import Form from '../components/Form';

describe('Footer', () => {
    const store = createStore(rootReducer, {});

    it('matches snapshot', () => {
        const component = Renderer.create(
            <Provider store={store}>
                <Form />
            </Provider>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});
