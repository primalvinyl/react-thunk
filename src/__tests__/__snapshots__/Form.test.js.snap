// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="screenName"
    onBlur={[Function]}
    onChange={[Function]}
    type="text"
  />
  <button
    disabled={false}
    type="submit"
  >
    Submit
  </button>
</form>
`;
